applications/PfemFluidDynamicsApplication/custom_strategies/strategies/two_step_v_p_strategy.h

```diff
@@ -828,14 +826,18 @@ class TwoStepVPStrategy : public SolvingStrategy<TSparseSpace,TDenseSpace,TLinea
 	    }else{
 	      ElementalVolume = 0;
 	    }
-	    double CriticalVolume=-10;
-	    if(ElementalVolume<CriticalVolume){
-	      (itElem)->Reset(ACTIVE);
+	    double ModelPartVolume=ModelerUtils.ComputeModelPartVolume(rModelPart);
+	    double CriticalVolume=0.01*ModelPartVolume/double(rModelPart.Elements().size());
```

Comment:
You are right. This is so because up to now we are working with homogenous meshes only. For handling with refined zones, the changes to do are several, included the one you pointed out.